test/WebJobs.Script.Tests/FileProvisioning/PowerShellFileProvisionerTests.cs

```diff
@@ -2,41 +2,189 @@
+
+            File.Exists(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+            File.Exists(Path.Combine(_scriptRootPath, ProfilePs1FileName));
+
+            string requirementsContent = File.ReadAllText(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+
+            const string ExpectedContent = @"# This file enables modules to be automatically managed by the Functions service.
+# See https://aka.ms/functionsmanageddependency for additional information.
+#
+@{
+    # For latest supported version, go to 'https://www.powershellgallery.com/packages/Az'. 
+    'Az' = '2.*'
+}
+";
+            Assert.Equal(ExpectedContent, requirementsContent, StringComparer.OrdinalIgnoreCase);
+
+            ValidateLogs(_loggerProvider, _scriptRootPath);
+        }
+
+        [Fact]
+        public async Task AugmentFiles_UnableToReachThePSGallery_Test()
+        {
+            EnsurePowerShellFilesDoNotExist(_scriptRootPath);
+
+            var powerShellFileProvisioner = new TestPowerShellFileProvisioner(_loggerFactory);
+            powerShellFileProvisioner.GetLatestAzModuleMajorVersionThrowsException = true;
             await powerShellFileProvisioner.ProvisionFiles(_scriptRootPath);
-            File.Exists(Path.Combine(_scriptRootPath, "requirements.psd1"));
-            File.Exists(Path.Combine(_scriptRootPath, "profile.ps1"));
+
+            File.Exists(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+            File.Exists(Path.Combine(_scriptRootPath, ProfilePs1FileName));
+
+            string requirementsContent = File.ReadAllText(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+
+            const string ExpectedContent = @"# This file enables modules to be automatically managed by the Functions service.
+# See https://aka.ms/functionsmanageddependency for additional information.
+#
+@{
+    # For latest supported version, go to 'https://www.powershellgallery.com/packages/Az'. Uncomment the next line and replace the MAJOR_VERSION, e.g., 'Az' = '2.*'
+    # 'Az' = 'MAJOR_VERSION.*'
+}
+";
+            Assert.Equal(ExpectedContent, requirementsContent, StringComparer.OrdinalIgnoreCase);
+
+            ValidateLogs(_loggerProvider, _scriptRootPath, unableToReachPSGallery: true);
         }
 
         [Theory]
         [InlineData(null)]
         [InlineData("")]
         public async Task AugmentFiles_EmptyScriptRootPath_Test(string scriptRootPath)
         {
-            var powerShellFileProvisioner = new PowerShellFileProvisioner();
+            var powerShellFileProvisioner = new PowerShellFileProvisioner(_loggerFactory);
             Exception ex = await Assert.ThrowsAsync<ArgumentException>(async () => await powerShellFileProvisioner.ProvisionFiles(scriptRootPath));
             Assert.True(ex is ArgumentException);
         }
+
+        [Fact]
+        public void FindLatestMajorVersionTest()
+        {
+            const string StreamContent = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<feed
```

Comment:
Nit: Move this to an embedded resource in test project

Reply:
>hmm it still is using LogInformation

Hi @pragnagopa: If you take a look at the latest iteration from Friday (https://github.com/Azure/azure-functions-host/pull/4782/files#diff-83876d5deb84b0328872cae4dd1bfde6), all logging calls go to ```_logger.LogDebug```.

Reply:
> Nit: Move this to an embedded resource in test project

Will do. Thank you for your comments!